extensions/pkg/controller/worker/genericactuator/actuator_reconcile.go

```diff
@@ -482,49 +475,11 @@ func (a *genericActuator) updateWorkerStatusMachineDeployments(ctx context.Conte
 		})
 	}
 
-	rollingUpdateCondition, err := buildRollingUpdateCondition(worker.Status.Conditions, isRollingUpdate)
-	if err != nil {
-		return err
+	patch := client.MergeFrom(worker.DeepCopy())
```

Comment:
Shouldn't we use a `StrategicMergePatch` here to make use of the list's `patchStrategy` /cc @timebertt

Reply:
There is no strategic merge patch for resources defined by CRDs :) 
ref https://github.com/gardener/gardener/blob/master/docs/development/kubernetes-clients.md#updates-patches-server-side-apply